developer-docs-site/static/move_demo/sources/CAPMessageBoard.move

```diff
@@ -0,0 +1,141 @@
+        move_to(account, notice_cap_container);
+        move_to(account, NoticeChangeEventHandle {
+            change_events:  Event::new_event_handle<NoticeChangeEvent>(account)
+        });
+        move_to(account, NoticeCapEventHandle {
+            change_events:  Event::new_event_handle<NoticeCapEvent>(account)
+        })
+
+    }
+
+    /// expose the function as a transaction interface
+    public(script) fun message_board_init(account: signer) {
+        message_board_init_internal(&account);
+    }
+
+    public fun view_message(board_addr: address) : vector<u8> acquires CapMessageBoard {
+        let board = borrow_global<CapMessageBoard>(board_addr);
+        board.notice
+    }
+
+    /// participant can register to
+    public fun register_internal(participant: &signer){
+        move_to(participant, NoticeChangeEventHandle {
+            change_events:  Event::new_event_handle<NoticeChangeEvent>(participant)
+        })
+    }
+
+    public(script) fun register(participant: signer){
+        register_internal(&participant);
+    }
+
+    /// board owner control adding new participants
+    public(script) fun add_participant(account: signer, participant: address) acquires NoticeCapEventHandle,NoticeChangeCapabilityContainer {
+        add_participant_internal(&account, participant);
+    }
+
+    public fun add_participant_internal(account: &signer, participant: address) acquires NoticeCapEventHandle, NoticeChangeCapabilityContainer {
+        let notice_cap_container = borrow_global_mut<NoticeChangeCapabilityContainer>(Signer::address_of(account));
+        notice_cap_container.addr = participant;
+        let event_handle = borrow_global_mut<NoticeCapEventHandle>(Signer::address_of(account));
+
+        Event::emit_event<NoticeCapEvent>(
+            &mut event_handle.change_events,
+            NoticeCapEvent {
+                timestamp: Timestamp::now_seconds(),
+                participant,
+                board: Signer::address_of(account)
+            }
+        );
+    }
+
+    public(script)  fun claim_notice_cap(account: signer, board: address) acquires NoticeChangeCapabilityContainer {
+        claim_notice_cap_internal( &account, board);
+    }
+
+    /// claim notice capability
+    public fun claim_notice_cap_internal(account: &signer, board: address) acquires NoticeChangeCapabilityContainer{
+        let notice_cap_container = borrow_global<NoticeChangeCapabilityContainer>(board);
+        assert!(notice_cap_container.addr == Signer::address_of(account), ACCOUNT_NO_NOTICE_CAP);
+        move_to(account, NoticeChangeCapability {is_active: true});
+    }
+
+    /// remove a participant from the ACL
+    public(script) fun remove_participant(account: signer, participant: address) acquires NoticeChangeCapability {
+        let cap = borrow_global_mut<NoticeChangeCapability>(participant);
+        assert!(Signer::address_of(&account) != participant, CANNOT_REMOVE_ADMIN_NOTICE_CAP);
+        cap.is_active = false;
```

Comment:
we should probably remove the cap instead of setting a bool, there's `move_from`

Reply:
was previously under the wrong impression that move_from must be called by a signer as move_to. But you are right. here, board_owner should be able to directly delete resource from participant's account